server/connection_manager.py

```diff
@@ -100,19 +85,19 @@ def update_code_cache(self, room_code: str, replace_data: ReplaceData) -> None:
                 self._rooms[room_code]["code"] = updated_code
 
     async def broadcast(
-        self, data: type[EventData], room_code: str, sender: Client | None = None, buggy: bool = False
+        self, data: EventResponse, room_code: str, sender: Client | None = None, buggy: bool = False
     ) -> None:
         """Broadcasts data to all active connections.
 
         Args:
             data: The data to be sent to the clients.
             room_code: The room to which the data will be sent.
-            sender (optional): The client who sent the message.
+            sender (optional): The client who sent the request.
             buggy (optional): To send back modified code.
         """
         if buggy:
             data = self._modify_code(room_code)
-            self.update_code_cache(room_code, data)
+            self.update_code_cache(room_code, data.data)
```

Comment:
`data.data` looks a bit weird. :laughing:

Reply:
It is what it is xD